main/uart_nic.c

```diff
@@ -263,20 +247,17 @@ static int IRAM_ATTR wifi_receive_cb(void *buffer, uint16_t len, void *eb) {
         }
     }
 
-    wifi_receive_buff *buff = malloc(sizeof(wifi_receive_buff));
-    if(!buff) {
-        goto cleanup;
-    }
-    buff->len = len;
-    buff->data = buffer;
-    buff->rx_buff = eb;
-    if (!xQueueSendToBack(uart_tx_queue, (void *)&buff, (TickType_t)0/*portMAX_DELAY*/)) {
-        free_wifi_receive_buff(buff);
+    struct uart0_tx_queue_item queue_item;
+    queue_item.header.type = MSG_PACKET_V2;
+    queue_item.header.up = 1; // if we are receiving data, we surely must be "up"
+    memcpy((uint8_t*)&queue_item.header.size, &len, 2);
+    queue_item.data = buffer;
+    if (xQueueSendToBack(uart0_tx_queue, &queue_item, 0) != pdTRUE) {
+        ESP_LOGE(TAG, "tx queue full");
+    } else {
+        return ESP_OK;
```

Comment:
We are sometimes returning a named constant (`ESP_OK`) and sometimes `0`. Can we be consistent? I know everything in C is technically an `int`, but my brain-internal type checker still suffers when seeing these things 😈

Reply:
About that... this is what the previous code does. I too am not comfortable with it. Maybe we should even send ESP_FAIL there.